Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp

```diff
@@ -486,6 +486,13 @@ namespace AzToolsFramework
 
                         if (oldName != newName)
                         {
+                            QRegExp rx("[_a-zA-Z0-9-\\s]+");
+                            if (!rx.exactMatch(newName.c_str()))
+                            {
+                                QMessageBox::information(NULL, "Title", "Special characters are not allowed", QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes);
```

Comment:
The message box should get parented to `AzToolsFramework::GetActiveWindow()` instead of `NULL`. Also, with this logic, the Yes/No choices on the message box don't change the outcome (it just gets replaced with the old name). So the message should instead clarify that is what will happen, or logic should be updated to use the choice to revert to old name or let the user continue editing.

Reply:
Suggested changes made.

Reply:
It looks like the logic still isn't using the result of the Yes/No. It should either state that the change won't be accepted with special characters and only have a single button (such as "OK"), or it should do something based on whether the user selected Yes or No.